gen/enum.go

```diff
@@ -21,11 +21,15 @@
 package gen
 
 import (
+	"fmt"
 	"strings"
 
 	"go.uber.org/thriftrw/compile"
 )
 
+// GoLabel is the annotation expected on enum items to override their label.
```

Comment:
similar to changelog, can we expand documentation here on what exactly this means and how this affects users relying on the JSON output from thriftrw-go

Reply:
how about:
```
// GoLabel is the annotation expected on enum items to override their label.
// Enum items tagged with `go.label=<TAGGED_NAME>` will generate go struct
// using `TAGGED_NAME` for JSON Marshal/Unmarshal
```

Reply:
There's no go struct involved for an enum, so I think it's a little confusing. I think it would be better as:
```
// GoLabel allows overriding the text formatting of an enum.
// Enum items will use the annotation value when serialized as
// a string. This affects String(), as well as text marshalling, used by JSON/YAML.
```